test/cpp/api/main.cpp

```diff
@@ -22,5 +23,8 @@ int main(int argc, char* argv[]) {
     session.configData().testsOrTags.emplace_back("~[cuda]");
   }
 
+  // Set a deterministic random seed for all tests.
+  torch::manual_seed(0);
```

Comment:
Is there really no way of running this in some sort of `setup()` method? Seeding only once at the beginning of all tests means that the random data will change if you execute only one test, and means that adding a test can cause another test to fail because the seed was perturbed. That's bad.

Reply:
well I can put it at the start of every single test that uses randomness if that's better

Reply:
With gtest we'll be able to put it in the fixture

Reply:
Good observation though, thanks. I guess putting it in every test that uses randomness is the best way for now.